crates/oxc_linter/src/rules/eslint/no_unused_vars/allowed.rs

```diff
@@ -298,4 +298,21 @@ impl NoUnusedVars {
             _ => false,
         }
     }
+
+    /// Returns `true` if this binding rest element should be allowed (i.e. not
+    /// reported). Currently, this handles the case where a rest element is part
+    /// of a TS function declaration.
+    pub(super) fn is_allowed_binding_rest_element<'a>(&self, symbol: &Symbol<'_, 'a>) -> bool {
+        for parent in symbol.iter_parents() {
+            // If this is a binding rest element that is part of a TS function parameter,
+            // for example: `function foo(...messages: string[]) {}`, then we will allow it.
+            if let AstKind::Function(f) = parent.kind() {
+                if f.is_typescript_syntax() {
+                    return true;
+                }
```

Comment:
i think we can just return f.is_typescript_syntax.

no need to continue up the tree if we've found a function

Reply:
^

Reply:
This also doesn't apply only to rest elements, it should apply to all parameters IMO.

Reply:
cool, thanks!